File: src/RangeSumQueryImmutable.java
public class RangeSumQueryImmutable {
    private static class NumArray {
        private final int[] result;

        public NumArray(int[] array) {
            result = new int[array.length + 1];
            for (int index = 0 ; index < array.length ; index++) {
                result[index + 1] = result[index] + array[index];
            }
        }

        public int sumRange(int i, int j) {
            return result[j] - result[i];
        }
    }
}

File: python/range_sum_query_immutable.py
from typing import List


class NumArray:

    def __init__(self, nums: List[int]):
        self.sum_till = [0] * (len(nums) + 1)
        for index, number in enumerate(nums):
            self.sum_till[index + 1] = self.sum_till[index] + number

    def sumRange(self, left: int, right: int) -> int:
        return self.sum_till[right + 1] - self.sum_till[left]
